src/client/automation/playback/press/utils.js

```diff
@@ -21,16 +23,19 @@ export function excludeShiftModifiedKeys (keyArray) {
 
             if (KEY_MAPS.shiftMap[nextKey])
                 keyArray[i + 1] = KEY_MAPS.shiftMap[nextKey];
+            else if (KEY_MAPS.reversedShiftMap[nextKey])
+                keysForKeyEvent[i + 1] = KEY_MAPS.reversedShiftMap[nextKey];
         }
 
         if (KEY_MAPS.shiftMap[key] && (!keyArray[i - 1] || keyArray[i - 1].toLowerCase() !== 'shift')) {
             keyArray[i] = KEY_MAPS.shiftMap[key];
             keyArray.splice(i, 0, 'shift');
+            keysForKeyEvent.splice(i, 0, 'shift');
             i++;
         }
     }
 
-    return keyArray;
+    return { keys: keyArray, keysForKeyEvent };
```

Comment:
This function became unclear now.
Before it was: `array -> excludeShiftModifiedKeys() -> array with some changes`
Not its: `array -> excludeShiftModifiedKeys() -> two arrays`.

Why it two? What the second array? It's unclear from the function name. So, think how to make it clean. (maybe change function's name, maybe do some other refactoring here)